lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -77,15 +82,43 @@ def fetch_bindnetloc
     Rex::Socket.to_authority(fetch_bindhost, fetch_bindport)
   end
 
+
   def generate(opts = {})
     opts[:arch] ||= module_info['AdaptedArch']
     opts[:code] = super
     @srvexe = generate_payload_exe(opts)
-    cmd = generate_fetch_commands
+    if datastore['FETCH_PIPE']
+      unless %w[WGET CURL].include?(datastore['FETCH_COMMAND'].upcase)
+        fail_with(Msf::Module::Failure::BadConfig, 'Unsupported Binary Selected for FETCH_PIPE option')
+      end
+      @pipe_cmd = generate_fetch_commands
+      vprint_status("Command served: #{@pipe_cmd}")
+      cmd = generate_pipe_command
+    else
+      cmd = generate_fetch_commands
+    end
     vprint_status("Command to run on remote host: #{cmd}")
     cmd
   end
 
+  def generate_pipe_command
+    # TODO: Make a check method that determines if we support a platform/server/command combination
+    if srvuri.length < 3
+      @pipe_uri = srvuri + 'p'
+    else
+      @pipe_uri = srvuri[...3]
+    end
+
+    case datastore['FETCH_COMMAND'].upcase
+    when 'WGET'
+      return _generate_wget_pipe
+    when 'CURL'
+      return _generate_curl_pipe
+    else
+      fail_with(Msf::Module::Failure::BadConfig, 'Unsupported Binary Selected for FETCH_PIPE option')
```

Comment:
```suggestion
      fail_with(Msf::Module::Failure::BadConfig, "Unsupported binary selected for FETCH_PIPE option: #{datastore['FETCH_COMMAND']}, must be WGET or CURL.")
```

Let's call out whats wrong and what it should be to be a touch more helpful.